crowdsource/tests/test_forms.py

```diff
@@ -0,0 +1,136 @@
+                                    'category': 'name',
+                                    'source': 'http://google.com',
+                                    })
+
+        self.assertTrue(form.is_valid(), form.errors)
+    
+    # Invalid jurisdicition
+    def test_resolverform_invalid_jurisdiction(self):
+        form = ResolverForm(data = {'jurisdiction':"notCreatedId",
+                                    'object_id':self.person.id,
+                                    'new_value':"new value",
+                                    'old_value':"old value",
+                                    'note':'Resolvers',
+                                    'reporter_name':'bikram',
+                                    'reporter_email':'bikram.bharti@gmail.com',
+                                    'category': 'name',
+                                    'source': 'http://google.com',
+                                    })
+
+        self.assertFalse(form.is_valid())
+    
+    # Invalid email
+    def test_resolverform_invalid_email(self):
+        form = ResolverForm(data = {'jurisdiction':self.jur.id,
+                                    'object_id':self.person.id,
+                                    'new_value':"new value",
+                                    'old_value':"old value",
+                                    'note':'Resolvers',
+                                    'reporter_name':'bikram',
+                                    'reporter_email':'bikram.bharti',
+                                    'category': 'name',
+                                    'source': 'http://google.com',
+                                    })
+
+        self.assertFalse(form.is_valid())
+
+    # Invalid url in source
+    def test_resolverform_invalid_url(self):
+        form = ResolverForm(data = {'jurisdiction':self.jur.id,
+                                    'object_id':self.person.id,
+                                    'new_value':"new value",
+                                    'old_value':"old value",
+                                    'note':'Resolvers',
+                                    'reporter_name':'bikram',
+                                    'reporter_email':'bikram.bharti@gmail.com',
+                                    'category': 'name',
+                                    'source': 'google',
+                                    })
+
+        self.assertFalse(form.is_valid())
+    
+    # Invalid category
+    def test_resolverform_invalid_category(self):
+        form = ResolverForm(data = {'jurisdiction':self.jur.id,
+                                    'object_id':self.person.id,
+                                    'new_value':"new value",
+                                    'old_value':"old value",
+                                    'note':'Resolvers',
+                                    'reporter_name':'bikram',
+                                    'reporter_email':'bikram.bharti@gmail.com',
```

Comment:
i'd avoid putting your real email here

Reply:
That's not real actually, but yes, I'll avoid adding my name from now on.